scripts/_check_docs_utils.py

```diff
@@ -0,0 +1,726 @@
+    if isinstance(node.exc, astroid.Name):
+        inferred = utils.safe_infer(node.exc)
+        if inferred:
+            excs = [inferred.name]
+    elif (isinstance(node.exc, astroid.Call) and
+          isinstance(node.exc.func, astroid.Name)):
+        target = utils.safe_infer(node.exc.func)
+        if isinstance(target, astroid.ClassDef):
+            excs = [target.name]
+        elif isinstance(target, astroid.FunctionDef):
+            for ret in target.nodes_of_class(astroid.Return):
+                if ret.frame() != target:
+                    # return from inner function - ignore it.
+                    continue
+
+                val = utils.safe_infer(ret.value)
+                if (val and isinstance(val, (
+                        astroid.Instance, astroid.ClassDef)) and
+                        utils.inherit_from_std_ex(val)):
+                    excs.append(val.name)
+    elif node.exc is None:
+        handler = node.parent
+        while handler and not isinstance(handler, astroid.ExceptHandler):
+            handler = handler.parent
+
+        if handler and handler.type:
+            inferred_excs = astroid.unpack_infer(handler.type)
+            excs = (exc.name for exc in inferred_excs
+                    if exc is not astroid.Uninferable)
+
+
+    try:
+        return set(
+            exc for exc in excs if not utils.node_ignores_exception(
+                node, exc))
+    except astroid.InferenceError:
+        return set()
+
+
+def docstringify(docstring):
+    for docstring_type in [SphinxDocstring, EpytextDocstring,
+                           GoogleDocstring, NumpyDocstring]:
+        instance = docstring_type(docstring)
+        if instance.is_valid():
+            return instance
+
+    return Docstring(docstring)
+
+
+class Docstring(object):
+    re_for_parameters_see = re.compile(r"""
+        For\s+the\s+(other)?\s*parameters\s*,\s+see
+        """, re.X | re.S)
+
+    supports_yields = None
+    """True if the docstring supports a "yield" section.
+
+    False if the docstring uses the returns section to document generators.
+    """
+
+    # These methods are designed to be overridden
+    # pylint: disable=no-self-use
+    def __init__(self, doc):
+        doc = doc or ""
+        self.doc = doc.expandtabs()
+
+    def is_valid(self):
+        return False
+
+    def exceptions(self):
+        return set()
+
+    def has_params(self):
+        return False
+
+    def has_returns(self):
+        return False
+
+    def has_rtype(self):
+        return False
+
+    def has_property_returns(self):
+        return False
+
+    def has_property_type(self):
+        return False
+
+    def has_yields(self):
+        return False
+
+    def has_yields_type(self):
+        return False
+
+    def match_param_docs(self):
+        return set(), set()
+
+    def params_documented_elsewhere(self):
+        return self.re_for_parameters_see.search(self.doc) is not None
+
+
+class SphinxDocstring(Docstring):
```

Comment:
Also, remove styles that we aren't using.

Reply:
I think we cannot do so since the classes are dependent on each other and removing one of them might break the linter.

Reply:
in that case, leave it as-is